user_tasks/signals.py

```diff
@@ -75,9 +75,17 @@ def _create_chain_entry(user_id, task_id, task_class, args, kwargs, callbacks, p
             if parent_name and not parent.name:
                 parent.set_name(parent_name)
     for callback in callbacks:
-        links = callback.options.get('link', [])
-        callback_class = import_string(callback.task)
-        _create_chain_entry(user_id, callback.id, callback_class, callback.args, callback.kwargs, links, parent=parent)
+        callback_class = import_string(callback['task'])
+
+        _create_chain_entry(
```

Comment:
```
(Pdb) callback
{'task': 'test_signals.sample_task', 'args': (1, '2'),
 'kwargs': {'user_task_name': 'Chain: 1, 2, 3'}, 
'options': {'task_id': '3fc0fef8-4090-40f6-b4df-d71e3a2add9e', 
'reply_to': 'e99174fa-afbd-379a-a239-3b67991f5115', 
'link': [test_signals.sample_task(1, '3')]}, 
'subtask_type': None, 'chord_size': None, 
'immutable': True}
```

Reply:
I did't find any documentation for this behaviour change.

Reply:
The string representation may have changed, but most of the `Signature` properties should be accessible either as `dict` entries or the properties defined at https://github.com/celery/celery/blob/master/celery/canvas.py#L525-L536 .  Did you hit an error that required the change?